File: JavaSolution/src/FaceBook/StrStr.java
package FaceBook;

/**
 * Created by xyunpeng on 1/18/16.
 */
public class StrStr {
    public int strStr(String haystack, String needle) {
        if (needle == null || haystack == null) {
            return  -1;
        }
        int hlength = haystack.length();
        int nlength = needle.length();
        for (int i = 0; i <= hlength - nlength; i++) {
            if (haystack.substring(i, i + nlength).equals(needle)) {
                return i;
            }
        }
        return -1;
    }
}

File: rust_solution/src/str_str.rs
pub struct Solution {}

impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        if needle.len() == 0 {
            return 0;
        }

        let mut i = 0;
        while i + needle.len() <= haystack.len() {
            if &haystack.as_bytes()[i..i + needle.len()] == needle.as_bytes() {
                return i as i32;
            } else {
                i += 1;
            }
        }

        -1
    }
}